### PR TITLE
UserStory #45 - edit business info

### PR DESCRIPTION
**Pull request by therobotisnotatoy**
Features #133 #134 #135 implemented.
Core style updated according to flake8 settings:
api/urls.py
api/views.py
api/permissions.py
beauty/urls.py

### CODE REVIEW

File: beauty/api/serializers/business_serializers.py [hunk 1]
```diff
@@ -1,14 +1,94 @@
...
+    """Serializer for specific business."""
+
+    owner_name = serializers.SerializerMethodField()
+    created_at = serializers.ReadOnlyField()
+    address = serializers.CharField(max_length=500)
+
+    def get_owner_name(self, obj):
+        """Return full name of business owner."""
+        full_name = f"{obj.owner.first_name} {obj.owner.last_name}"
+        logger.debug("Owner name transferred to frontend")
```
Comment (morento101) [inline beauty/api/serializers/business_serializers.py]: full_name = obj.owner.get_full_name()

File: beauty/api/serializers/business_serializers.py [hunk 2]
```diff
@@ -1,14 +1,94 @@
...
-        fields = ('name', 'type', 'owner', 'description')
+        fields = ("name", "business_type", "owner", "description")
+
+    def validate_owner(self, value):
+        """User validation.
+
+        Checks if such user exists and validates if user belongs to
+        Specialist group
+        """
+        group = value.groups.filter(name="Owner").first()
```
Comment (Misha86) [inline beauty/api/serializers/business_serializers.py]: use try-except  block and .get() method (expert spoke about it)

File: beauty/api/serializers/business_serializers.py [hunk 3]
```diff
@@ -1,14 +1,94 @@
...
+            raise ValidationError(
+                _("Only Owners can create Business"), code="invalid",
+            )
+
+        return value
 
     def to_representation(self, instance):
+        """Display owner full name."""
         data = super().to_representation(instance)
         owner = CustomUser.objects.filter(id=data["owner"]).first()
```
Comment (Misha86) [inline beauty/api/serializers/business_serializers.py]: use .get() method

File: beauty/api/serializers/business_serializers.py [hunk 4]
```diff
@@ -1,14 +1,94 @@
...
+
+
+class BusinessAllDetailSerializer(serializers.ModelSerializer):
+    """Serializer for specific business."""
+
+    owner_name = serializers.SerializerMethodField()
+    created_at = serializers.ReadOnlyField()
+    address = serializers.CharField(max_length=500)
+
+    def get_owner_name(self, obj):
```
Comment (Misha86) [inline beauty/api/serializers/business_serializers.py]: use to_representation() method.

File: beauty/api/serializers/business_serializers.py [hunk 5]
```diff
@@ -1,14 +1,94 @@
...
+        """Meta for OwnerBusinessesSerializer class."""
+
+        model = Business
+        fields = ("business_url", "owner_url", "name", "business_type", "address")
+
+
+class BusinessAllDetailSerializer(serializers.ModelSerializer):
+    """Serializer for specific business."""
+
+    owner_name = serializers.SerializerMethodField()
```
Comment (Misha86) [inline beauty/api/serializers/business_serializers.py]: if you are using the to_representation() method you can delete this code string

**Review by Misha86 — CHANGES_REQUESTED** [review]

File: beauty/api/serializers/business_serializers.py (hunk 2 above)
Comment (therobotisnotatoy) [reply beauty/api/serializers/business_serializers.py]: done

File: beauty/api/serializers/business_serializers.py (hunk 3 above)
Comment (therobotisnotatoy) [reply beauty/api/serializers/business_serializers.py]: done

File: beauty/api/serializers/business_serializers.py (hunk 5 above)
Comment (therobotisnotatoy) [reply beauty/api/serializers/business_serializers.py]: done

File: beauty/api/serializers/business_serializers.py (hunk 1 above)
Comment (therobotisnotatoy) [reply beauty/api/serializers/business_serializers.py]: refactored according to others comments

File: beauty/api/serializers/business_serializers.py (hunk 4 above)
Comment (therobotisnotatoy) [reply beauty/api/serializers/business_serializers.py]: done

**Review by OleksiiDatsiuk — APPROVED** [review]

**Review by Misha86 — APPROVED** [review]

**Pull request merged**